cmd/crates/soroban-spec-typescript/ts-tests/src/test.ts

```diff
@@ -0,0 +1,188 @@
+  t.deepEqual(await contract.simple({ simple }), simple)
+})
+
+test('complex with struct', async t => {
+  const arg = { tag: 'Struct', values: [{ a: 0, b: true, c: 'hello' }] } as const
+  const ret = { tag: 'Struct', values: { a: 0, b: true, c: 'hello' } }
+  t.deepEqual(await contract.complex({ complex: arg }), ret)
+})
+
+test('complex with tuple', async t => {
+  const arg = { tag: 'Tuple', values: [[{ a: 0, b: true, c: 'hello' }, { tag: 'First', values: undefined }]] } as const
+  const ret = { tag: 'Tuple', values: [{ a: 0, b: true, c: 'hello' }, ['First']] }
+  t.deepEqual(await contract.complex({ complex: arg }), ret)
+})
+
+test('complex with enum', async t => {
+  const arg = { tag: 'Enum', values: [{ tag: 'First', values: undefined }] } as const
+  const ret = { tag: 'Enum', values: ['First'] }
+  t.deepEqual(await contract.complex({ complex: arg }), ret)
+})
+
+test('complex with asset', async t => {
+  const arg = { tag: 'Asset', values: [publicKey, 1n] } as const
+  const ret = { tag: 'Asset', values: publicKey }
+  t.deepEqual(await contract.complex({ complex: arg }), ret)
+})
+
+test('complex with void', async t => {
+  const complex = { tag: 'Void', values: undefined } as const
+  t.deepEqual(await contract.complex({ complex }), complex)
+})
+
+test('addresse', async t => {
+  t.is(await contract.addresse({ addresse: publicKey }), publicKey)
+})
+
+test('bytes', async t => {
+  const bytes = Buffer.from('hello')
+  t.deepEqual(await contract.bytes({ bytes }), bytes)
+})
+
+test.failing('bytes_n', async t => {
+  const bytes_n = Buffer.from('1') // what's the correct way to construct bytes_n?
+  t.deepEqual(await contract.bytesN({ bytes_n }), bytes_n)
+})
+
+test.failing('card', async t => {
+  const card = 11
+  t.is(await contract.card({ card }), card)
+})
+
+test('boolean', async t => {
+  t.is(await contract.boolean({ boolean: true }), true)
+})
+
+test('not', async t => {
+  t.is(await contract.not({ boolean: true }), false)
+})
+
+test('i128', async t => {
+  t.is(await contract.i128({ i128: -1n }), -1n)
+})
+
+test('u128', async t => {
+  t.is(await contract.u128({ u128: 1n }), 1n)
+})
+
+test('multi_args', async t => {
+  t.is(await contract.multiArgs({ a: 1, b: true }), 1)
+  t.is(await contract.multiArgs({ a: 1, b: false }), 0)
+})
+
+test('map', async t => {
+  const map = new Map()
+  map.set(1, true)
+  map.set(2, false)
+  // map.set(3, 'hahaha') // should throw an error
+  t.deepEqual(await contract.map({ map }), Object.fromEntries(map.entries()))
+})
+
+test('vec', async t => {
+  const vec = [1, 2, 3]
+  t.deepEqual(await contract.vec({ vec }), vec)
+})
+
+test('tuple', async t => {
+  const tuple = ['hello', 1] as const
+  t.deepEqual(await contract.tuple({ tuple }), tuple)
+})
+
+test.failing('option', async t => {
+  // this makes sense
+  t.deepEqual(await contract.option({ option: 1 }), 1)
+
+  // this passes but shouldn't
+  t.deepEqual(await contract.option({ option: undefined }), 0)
```

Comment:
can/should we make this an intentional test failure?

Reply:
This is changed in my updated tests that use the new ContractSpec class to handle the parsing.

Reply:
It returns `undefined`.

Reply:
And this already is an intentional failure; see the `t.failing` above. But it doesn't fail until the line below, when we get to the `assert returns undefined` line, and I wanted to clearly comment on the unexpected behavior of options.